### PR TITLE
make vpc_nat_gateway redo eip fip snat dnat iptables rules after k8s cluster reboot

### PR DESCRIPTION
**Pull request by wenwenxiong**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
Examples of user facing changes:
- Features
- Bug fixes
- Docs
- Tests

### Which issue(s) this PR fixes:
Fixes #(https://github.com/kubeovn/kube-ovn/issues/3241)

### <samp>🤖 Generated by Copilot at b54535b</samp>

This pull request improves the reliability and consistency of the NAT gateway controller and the FIP feature. It adds logic to handle pod restarts and redo iptables rules for EIP, FIP, DNAT, and SNAT objects. It also changes the default value of the `pod-default-fip-type` flag to "iptables" and adds a function to initialize the VPC NAT gateways.

### <samp>🤖 Generated by Copilot at b54535b</samp>

> _Sing, O Muse, of the skillful coder who changed the flag_
> _Of `pod-default-fip-type` to match the docs and logic,_
> _And who added to the controller the function `initVpcNatGw`_
> _To initialize the gateways of the clouds, like Zeus the thunderer._

### <samp>🤖 Generated by Copilot at b54535b</samp>

*  Change the default value of the `pod-default-fip-type` flag to "iptables" to match the documentation and code logic ([link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-b47da0cc17fe8ee9178f1d883fbc42f45187f2adbb02d05ca9dd8644079bee75L153-R153))
*  Add a function call to initialize the VPC NAT gateways in the controller's `initResourceOnce` function ([link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-243ebed2765f75e6a54f57167212fefb08c3b2a85967ad2acbc0eb78919019c1R1164-R1166))
*  Implement the `initVpcNatGw` function to list and enqueue the VPC NAT gateways for initialization and check their annotations ([link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-41b4c692987143b2796e2664174ba6858e7c51329d823c612556ef6cd32185c4R1095-R1124))
*  Import the `time` package to parse and compare timestamps for the EIP, FIP, DNAT, and SNAT configurations ([link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-c2fca6de8b38cf14047d762d1029e969197a5fd0d5172062dce753f1e9613c92R8))
*  Add logic to check the redo time and start time of the EIP, FIP, DNAT, and SNAT configurations in the `handleAddIptablesEip`, `handleAddIptablesFip`, `handleAddIptablesDnatRule`, and `handleAddIptablesSnatRule` functions, respectively, and reconfigure them if needed ([link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-c2fca6de8b38cf14047d762d1029e969197a5fd0d5172062dce753f1e9613c92L217-R227), [link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-9730fd87f1276f8f297415b3bbf5407de631488b88c318d7fd500b0b6a5a9c4aL496-R499), [link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-9730fd87f1276f8f297415b3bbf5407de631488b88c318d7fd500b0b6a5a9c4aR518-R530), [link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-9730fd87f1276f8f297415b3bbf5407de631488b88c318d7fd500b0b6a5a9c4aL684-R700), [link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-9730fd87f1276f8f297415b3bbf5407de631488b88c318d7fd500b0b6a5a9c4aR715-R726), [link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-9730fd87f1276f8f297415b3bbf5407de631488b88c318d7fd500b0b6a5a9c4aL857-R885), [link](https://github.com/kubeovn/kube-ovn/pull/3267/files?diff=unified&w=0#diff-9730fd87f1276f8f297415b3bbf5407de631488b88c318d7fd500b0b6a5a9c4aR897-R908))

### CODE REVIEW

File: pkg/controller/vpc_nat_gw_nat.go [hunk 1]
```diff
@@ -493,10 +493,10 @@ func (c *Controller) handleAddIptablesFip(key string) error {
 		klog.Error(err)
 		return err
 	}
-	if fip.Status.Ready && fip.Status.V4ip != "" {
+	/*if fip.Status.Ready && fip.Status.V4ip != "" {
```
Comment (bobz965) [inline pkg/controller/vpc_nat_gw_nat.go:496]: 我在想重建的逻辑是不是可以完全放到handleUpdate<nat> 方法里面，可以不动handleAdd接口么?

Comment (oilbeater) [inline pkg/controller/vpc_nat_gw_nat.go:496]: 注释的代码可以直接删除，不需要保留，后面几段代码也是一样的

**Review by oilbeater — CHANGES_REQUESTED** [review]

**Comment by oilbeater** [comment]
@wenwenxiong hi, are you still working on this pr?

**Comment by bobz965** [comment]
> @wenwenxiong hi, are you still working on this pr?

if not, I will go on finish this pr.

**Comment by wenwenxiong** [comment]
i am not to do with this pr now

**Review by bobz965 — APPROVED** [review]

**Review by oilbeater — APPROVED** [review]

**Pull request merged**